src/meta/meta_bulk_load_service.cpp

```diff
@@ -769,17 +779,45 @@ void bulk_load_service::update_partition_status_on_remote_storage(const std::str
                                                                   bulk_load_status::type new_status,
                                                                   bool should_send_request)
 {
-    zauto_read_lock l(_lock);
+    zauto_write_lock l(_lock);
     partition_bulk_load_info pinfo = _partition_bulk_load_info[pid];
-
-    if (pinfo.status == new_status) {
+    if (pinfo.status == new_status && new_status != bulk_load_status::BLS_DOWNLOADING) {
```

Comment:
This if statement implies DOWNLOADING => DOWNLOADING requires a meta sync. I don't understand the reason of this code. I suggest a comment here.

Reply:
When trying to call `update_partition_status_on_remote_storage` to update partition status to downloading, it will only be happened when rolling back to downloading. Consider the case, app is downloading, partition[0-2] is downloaded, partition[3] is downloading, some recoverable error happens, whole status will rollback to downloading, app status downloading -> downloading, partition status should also turn to downloading,  and clear local status in function `update_partition_status_on_remote_storage_reply`